### ISSUE TITLE
runtime error: slice bounds out of range [-1:]

### ISSUE THREAD

**Issue by ryanpcmcquen** [chriswalz/bit#13]
I get this on some of my git repos:
```
λ bit
runtime error: slice bounds out of range [-1:]
goroutine 1 [running]:
runtime/debug.Stack(0x12aeb60, 0xc0000b4008, 0xc0000d76f0)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/chriswalz/bit/cmd.HandleExit()
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.2/cmd/util.go:341 +0xc5
panic(0x1228560, 0xc0001a2000)
	/usr/local/go/src/runtime/panic.go:969 +0x175
github.com/chriswalz/bit/cmd.BranchList(0x5d, 0x4f, 0x4f)
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.2/cmd/util.go:179 +0x550
github.com/chriswalz/bit/cmd.BranchListSuggestions(0x13e2980, 0xc000122000, 0x5d)
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.2/cmd/util.go:208 +0x34
github.com/chriswalz/bit/cmd.CreateSuggestionMap(0x13e2980, 0xc0000d7c80, 0x1172357)
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.2/cmd/rootShell.go:53 +0x88
github.com/chriswalz/bit/cmd.glob..func3(0x13e2980, 0x1416ba0, 0x0, 0x0)
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.2/cmd/rootShell.go:20 +0x45
github.com/spf13/cobra.(*Command).execute(0x13e2980, 0xc0000961b0, 0x0, 0x0, 0x13e2980, 0xc0000961b0)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x13e2980, 0x11d53a5, 0xc000100000, 0xc000108000)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/chriswalz/bit/cmd.Execute()
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.2/cmd/rootShell.go:76 +0x31
main.bitcli(...)
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.2/main.go:55
main.main()
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.2/main.go:43 +0x18b
```

**Comment by chriswalz** [comment]
Let me know if there's anything special about the commit dates for those repo's as I believe it's something related to that. 

I pushed a hacky-ish work around for the out of bounds issue

**Comment by ryanpcmcquen** [comment]
@chriswalz, here are some examples:
```
commit e5cffc55bcc4acf34b1be3333018b6486044164d (HEAD -> release-v2.11.0, origin/release-v2.11.0)
Merge: ef5ee40 2f41d5e
Date:   Fri Oct 2 15:19:46 2020 +0000

commit 2f41d5ea185af9ba14f4458523eeda7e9075351c (origin/feature_FD-5860, feature_FD-5860)
Date:   Thu Oct 1 17:22:49 2020 -0700

commit ef5ee4050ff6cc4e1b05d695200fea0ac366424b
Merge: 90c2592 e132131
Date:   Wed Sep 23 23:06:22 2020 +0000

commit e13213113727167623e26f9967b85703ba8e8998 (origin/feature_FD-5769, feature_FD-5769)
Date:   Wed Sep 23 16:02:38 2020 -0700

commit 1263c5c059a7dcf33d56cb7e052b9a74daab2505
Merge: 50ef579 90c2592
Date:   Wed Sep 23 22:25:17 2020 +0000

commit 50ef579ac067e3b0bcee925324637f4a7032fb5e
Date:   Wed Sep 23 15:24:49 2020 -0700
```

**Comment by ryanpcmcquen** [comment]
Oldest commit:
```
commit 3221f6cd41502b33bbd552cdb338a08cf193ba8b
Date:   Mon Jan 22 16:49:24 2018 -0800
```

**Comment by ryanpcmcquen** [comment]
The issue persists with 0.5.3:
```
λ bit
runtime error: index out of range [1] with length 1
goroutine 1 [running]:
runtime/debug.Stack(0x12aeba0, 0xc000130008, 0xc0001556f0)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/chriswalz/bit/cmd.HandleExit()
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.3/cmd/util.go:348 +0xc5
panic(0x1228580, 0xc000218000)
	/usr/local/go/src/runtime/panic.go:969 +0x175
github.com/chriswalz/bit/cmd.BranchList(0x5d, 0x4f, 0x4f)
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.3/cmd/util.go:184 +0x565
github.com/chriswalz/bit/cmd.BranchListSuggestions(0x13e2980, 0xc0001a2000, 0x5d)
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.3/cmd/util.go:215 +0x34
github.com/chriswalz/bit/cmd.CreateSuggestionMap(0x13e2980, 0xc000155c80, 0x1172357)
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.3/cmd/rootShell.go:53 +0x88
github.com/chriswalz/bit/cmd.glob..func3(0x13e2980, 0x1416ba0, 0x0, 0x0)
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.3/cmd/rootShell.go:20 +0x45
github.com/spf13/cobra.(*Command).execute(0x13e2980, 0xc0001101b0, 0x0, 0x0, 0x13e2980, 0xc0001101b0)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x13e2980, 0x11d53a5, 0xc00017e000, 0xc000186000)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/chriswalz/bit/cmd.Execute()
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.3/cmd/rootShell.go:76 +0x31
main.bitcli(...)
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.3/main.go:55
main.main()
	/go/pkg/mod/github.com/chriswalz/bit@v0.5.3/main.go:43 +0x18b
```

**Comment by chriswalz** [comment]
So this command is used behind the scenes

git for-each-ref --sort=-committerdate refs/heads/, refs/remotes --format='%(authordate:short); %(authorname); %(color:red)%(objectname:short); %(color:yellow)%(refname:short)%(color:reset); (%(color:green)%(committerdate:relative)%(color:reset))'

If you could show me the difference between a failing vs working repo that'd be perfect

**Comment by ryanpcmcquen** [comment]
Broken:
```
λ git for-each-ref --sort=-committerdate refs/heads/, refs/remotes --format='%(authordate:short); %(authorname); %(color:red)%(objectname:short); %(color:yellow)%(refname:short)%(color:reset); (%(color:green)%(committerdate:relative)%(color:reset))'
warning: ignoring broken ref refs/remotes/origin/HEAD
'2020-10-02; Ryan McQuen; e5cffc5; origin/release-v2.11.0; (7 days ago)'
'2020-10-01; Ryan McQuen; 2f41d5e; origin/feature_FD-5860; (8 days ago)'
'2020-09-23; Ryan McQuen; e132131; origin/feature_FD-5769; (2 weeks ago)'
'2020-09-22; Ryan McQuen; c3c6616; origin/master; (2 weeks ago)'
'2020-09-16; Ryan McQuen; 6279706; origin/release-v2.10.0; (3 weeks ago)'
'2020-09-16; Ryan McQuen; 0830b84; origin/bug_FD-5681; (3 weeks ago)'
```
Working:
```
λ git for-each-ref --sort=-committerdate refs/heads/, refs/remotes --format='%(authordate:short); %(authorname); %(color:red)%(objectname:short); %(color:yellow)%(refname:short)%(color:reset); (%(color:green)%(committerdate:relative)%(color:reset))'
'2020-10-06; Ryan McQuen; bf84c09; origin/HEAD; (3 days ago)'
'2020-10-06; Ryan McQuen; bf84c09; origin/master; (3 days ago)'
'2020-10-06; Ryan McQuen; a302055; origin/code_click_issue; (3 days ago)'
```

**Comment by ryanpcmcquen** [comment]
Somehow HEAD got jacked, so I fixed that warning with this:
```
git remote set-head origin master
```
And now `bit` works as expected, _but_, maybe you want to handle that warning more gracefully?

**Comment by chriswalz** [comment]
fixed with 09c6ea141f7e26ec89ab335e6b60bbc343ba7126
